spec/lib/vanagon/engine/pooler_spec.rb

```diff
@@ -21,28 +21,86 @@
   before :each do
     # suppress `#warn` output during tests
     allow_any_instance_of(Vanagon::Platform::DSL).to receive(:warn)
+
+    # stubbing ENV doesn't actually intercept ENV because ENV
```

Comment:
Nope. That only works at what I'm affectionately referring to as the "Ruby" level. Anything that reads environment variables from what I'm frustratingly referring to as the "C" level will completely disregard that and return your real, true process environment. So when the code paths we're testing call `File.expand_path`, that's a pure C method and it returns a value from the C level.

```ruby
>> ENV.inspect
=> {"HOME"=>"/var/folders/jg/0fyykqj50wgc48f_6bygnblr0000gn/T/d20170105-47621-yontgt"}
>> puts File.join(ENV['HOME']), .vmfloaty.yml)
=> /var/folders/jg/0fyykqj50wgc48f_6bygnblr0000gn/T/d20170105-47621-yontgt/.vmfloaty.yml
>> puts File.expand_path('~/.vmfloaty.yml')
=> /Users/mckern/.vmfloaty.yml
>>
```